smartsim/_core/launcher/dragon/dragonBackend.py

```diff
@@ -67,34 +66,48 @@
 logger = get_logger(__name__)
 
 
-class DragonStatus(str, Enum):
-    ERROR = str(dragon_group_state.Error())
-    RUNNING = str(dragon_group_state.Running())
-
-    def __str__(self) -> str:
-        return self.value
-
-
 @dataclass
 class ProcessGroupInfo:
     status: SmartSimStatus
     """Status of step"""
     process_group: t.Optional[dragon_process_group.ProcessGroup] = None
     """Internal Process Group object, None for finished or not started steps"""
-    puids: t.Optional[t.List[t.Optional[int]]] = None  # puids can be None
-    """List of Process UIDS belonging to the ProcessGroup"""
-    return_codes: t.Optional[t.List[int]] = None
-    """List of return codes of completed processes"""
     hosts: t.List[str] = field(default_factory=list)
     """List of hosts on which the Process Group """
     redir_workers: t.Optional[dragon_process_group.ProcessGroup] = None
     """Workers used to redirect stdout and stderr to file"""
 
     @property
-    def smartsim_info(self) -> t.Tuple[SmartSimStatus, t.Optional[t.List[int]]]:
+    def smartsim_info(self) -> t.Tuple[SmartSimStatus, t.List[int]]:
         """Information needed by SmartSim Launcher and Job Manager"""
         return (self.status, self.return_codes)
 
+    @property
+    def puids(self) -> t.List[int]:
+        """List of Process UIDS belonging to the ProcessGroup"""
+        return list(set(itertools.chain(self.active_puids, self.inactive_puids)))
+
+    @property
+    def active_puids(self) -> t.List[int]:
+        if self.process_group is None:
+            return []
+        return list(self.process_group.puids)
+
+    @property
+    def inactive_puids(self) -> t.List[int]:
+        if self.process_group is None:
+            return []
+        return [puid for puid, _ in self.process_group.inactive_puids]
+
+    @property
+    def return_codes(self) -> t.List[int]:
+        """List of return codes of completed processes"""
+        if self.process_group is None:
+            return [-1]
```

Comment:
should -1 and -9 be enshrined somewhere (enum, constant) so they have a descriptive name? Where do -1 and -9 come from?

Reply:
> should -1 and -9 be enshrined somewhere (enum, constant) so they have a descriptive name?

Probably, hahahaha. I can add one!

> Where do -1 and -9 come from?

Honestly not sure. I used them for backward compatibility reasons since that was what they were manually set to when a group info was created w/o a process group and when a process group was manually canceled previously. @al-rigazzi might have some better insight here?

Reply:
I think we can se the standard System Error Codes?

Reply:
We still need the `[-1]` in the case were no process group is actually launched when creating a `ProcessGroupInfo` ([see here](https://github.com/CrayLabs/SmartSim/blob/c8f6decf36f9c1a148c1ae8739e67abcc82e7d5c/smartsim/_core/launcher/dragon/dragonBackend.py#L707-L715)), but we can 100% use the status codes in the case of a process group being cancelled! Will remove!